compiler/AST/UseStmt.cpp

```diff
@@ -508,8 +522,20 @@ void UseStmt::trackMethods() {
   }
 }
 
-ModuleSymbol* UseStmt::checkIfModuleNameMatches(const char* name) {
-  if (SymExpr* se = toSymExpr(src)) {
+Symbol* UseStmt::checkIfModuleNameMatches(const char* name) {
+  if (isARename()) {
+    // Use statements that rename the module should only allow us to find the
+    // new name, not the original one.
+    if (strcmp(name, getRename()) == 0) {
```

Comment:
I'm surprised to see strcmp here because astr should allow pointer comparison. Could you try doing the changes above and then just `==` for  pointer comparison here, and add it to your further work to investigate if that causes problems?